### ISSUE TITLE
Proxmox 7 - Script does not work

### ISSUE THREAD

**Issue by Mansaylon** [whiskerz007/proxmox_hassio_lxc#45]
Hello
New Proxmox 7 (7.0.11) installed. Still, the script doesn't work.

root@pve:~# bash -c "$(wget -qLO - https://github.com/whiskerz007/proxmox_hassio_lxc/raw/master/create_container.sh)"
[INFO] Using 'local' for template storage.
[INFO] Using 'local-lvm' for container storage.
Updating LXC template list...
Downloading LXC template...
Creating LXC container...
[INFO] LXC container '100' was successfully created.
Starting LXC container...
run_buffer: 316 Script exited with status 1
lxc_init: 816 Failed to run lxc.hook.pre-start for container "100"
__lxc_start: 2007 Failed to initialize container "100"
startup for container '100' failed
**[ERROR] 255@100 Unknown failure occured.**
  Logical volume "vm-100-disk-0" successfully removed

**Comment by gdreelin** [comment]
Yeah I got the same thing when trying to rebuild a new container.

![Untitled4](https://user-images.githubusercontent.com/802649/134408038-432c892f-a3de-4c55-912b-730eecee4b15.png)

**Comment by tiagofreire-pt** [comment]
Use this script version instead:

https://raw.githubusercontent.com/whiskerz007/proxmox_hassio_lxc/c4ba4adc75c77bd192c60ad6daf3f57d5ff7f3cd/create_container.sh

**Comment by ghost** [comment]
@tiagofreire-pt thanks.. that script worked

**Comment by FordMustang0288** [comment]
After running this script, I can't access the configuration site in the browser. Under proxmox networking it shows the container is set to static IP and has a mac address. I've also tried DHCP to no avail.

**Comment by quenthal** [comment]
It could simply be, that it just starts extremely slowly. More than often LXC hassio get stuck in "waiting for..." if you check from console.

Even though I have managed to install it in Proxmox 7, it is much slower to start than previously.

**Comment by Hanselljlh** [comment]
> Use this script version instead:
> 
> https://raw.githubusercontent.com/whiskerz007/proxmox_hassio_lxc/c4ba4adc75c77bd192c60ad6daf3f57d5ff7f3cd/create_container.sh

Still end up with an error @ line 131

**Comment by Maltavius** [comment]
> After running this script, I can't access the configuration site in the browser. Under proxmox networking it shows the container is set to static IP and has a mac address. I've also tried DHCP to no avail.

I have the same issue, the script reports an IP and port but that IP and port isn't working because network is specified to static with no information in Proxmox.

When I log into the lxc console and check if docker is running it has the supervisor image but nothing is running at all so even if I get the IP to work there isn't any homeassistant there to connect to

**Comment by Cooper81** [comment]
yes.
same with me

**Comment by Hanselljlh** [comment]
Any updates?

**Comment by fragtion** [comment]
Same problem -
```
run_buffer: 316 Script exited with status 255
lxc_init: 816 Failed to run lxc.hook.pre-start for container "123"
__lxc_start: 2007 Failed to initialize container "123"
TASK ERROR: startup for container '123' failed
```
My proxmox repo was missing some entries, after updating the system everything works fine again

**Comment by Hanselljlh** [comment]
> My proxmox repo was missing some entries, after updating the system everything works fine again

What were you missing?

I am still getting line 130 for my error.

**Comment by fragtion** [comment]
> > My proxmox repo was missing some entries, after updating the system everything works fine again
> 
> What were you missing?
> 
> I am still getting line 130 for my error.

First make sure you're on bullseye
`sed -i 's/buster\/updates/bullseye-security/g;s/buster/bullseye/g' /etc/apt/sources.list`

I added this repo:
`echo deb [arch=amd64] http://download.proxmox.com/debian/pve bullseye pve-no-subscription > /etc/apt/sources.list.d/pve-install-repo.list`

Then ran:
`apt clean && apt update && apt dist-upgrade`

A bunch of updates showed up, which I agreed to upgrade... now I'm on proxmox v7.0-13, and the container magically works fine again. I think the package that fixed it was either pve-firmware, or the pve linux kernel but I'm not sure on that.

You might also need to get `/etc/apt/sources.list.d/trusted.gpg.d/proxmox-release-bullseye.gpg` if you don't have it already, in order for that new repo to work

**Comment by zfrank2601** [comment]
I also got 130 error

```
root@pve:~# bash -c "$(wget -qLO - https://raw.githubusercontent.com/whiskerz007/proxmox_hassio_lxc/c4ba4adc75c77bd192c60ad6daf3f57d5ff7f3cd/create_container.sh)"
[INFO] Using 'sd120gbdiv' for template storage.
[INFO] Using 'nvme0n1' for container storage.
Updating LXC template list...
Creating LXC container...
[INFO] LXC container '121' was successfully created.
Starting LXC container...
Setting up container OS...
Updating container OS...
Installing prerequisites...
Installing Docker...
Configuring Docker...
[ERROR] 8@130 Unknown failure occured.
```
I checked repos and also updated the complete system
now on proxmox v7.0-14

**Comment by Hanselljlh** [comment]
On v7. 0-14+1 and still failing to install at line 130.

**Comment by ghost** [comment]
> It could simply be, that it just starts extremely slowly. More than often LXC hassio get stuck in "waiting for..." if you check from console.
> 
> Even though I have managed to install it in Proxmox 7, it is much slower to start than previously.

I see the same thing.. sometimes I just sits there.. once I open the console in prox it starts up after a few min and everything starts to work

**Comment by Hanselljlh** [comment]
How are you guys even getting to that point? When we error on line 130 of the initial install it deletes the container it started to create.

**Comment by ghost** [comment]
@Hanselljlh I installed a while ago.. but I just did some looking and it seems that line 130 is trying to load a file from the web that no longer exists.. and it crashes

**Comment by Hanselljlh** [comment]
Is there a way to fix that?

**Comment by ghost** [comment]
yes.. debug the problem and create a new file that works :)

I have created a file that I won't maintain.. links to files have all been moved around, so I have spent some time finding their new locations and editing the original file from whiskerz007 to fix the problems

just run `bash -c "$(wget -qLO - https://raw.githubusercontent.com/oziee/proxmox-ha/main/ha-install.sh)"`

```
root@pve:~# bash -c "$(wget -qLO - https://raw.githubusercontent.com/oziee/proxmox-ha/main/ha-install.sh)"
[INFO] Using 'local' for template storage.
[INFO] Using 'local-lvm' for container storage.
Updating LXC template list...
Creating LXC container...
[INFO] LXC container '113' was successfully created.
Starting LXC container...
Setting up container OS...
Updating container OS...
Installing prerequisites...
Installing Docker...
Configuring Docker...
Configuring NetworkManager...
Creating Home Assistant config...
Downloading Home Assistant Supervisor container...
Installing Home Assistant Supervisor...
Creating service to fix boot time check...
Starting Home Assistant...
Installing the 'ha' cli...
Configuring 'ha' cli prompt...
Cleanup...
[INFO] Successfully created Home Assistant LXC to 113.


Home Assistant is reachable by going to the following URLs.

      http://10.0.1.225:8123
      http://pve.local:8123


root@pve:~# 
```

**Comment by Hanselljlh** [comment]
Now it's failing at line 132. 

Failing on the docker setup. 

I really appreciate your help. Wouldn't be mad if you stop here lol. I sort of know how to read the script but I wouldn't know where/how to fix the files.

**Comment by zfrank2601** [comment]
For me it´s working
My promox-version:

Kernelversion Linux 5.11.22-7-pve #1 SMP PVE 5.11.22-12 (Sun, 07 Nov 2021 21:46:36 +0100)
PVE-Manager-Version pve-manager/7.0-14/a9dbe7e3

**Comment by Hanselljlh** [comment]
Kernel Version Linux 5.11.22-7-pve #1 SMP PVE 5.11.22-12 (Sun, 07 Nov 2021 21:46:36 +0100)
PVE Manager Version pve-manager/7.0-14+1/08975a4c

Is the 14+1 going to make. A difference?

And just to make sure I'm not just screwing this up because of something simple. I run the bash command in the shell or ssh to the root of the prox server.

**Comment by zfrank2601** [comment]
I tested again with 14+1, also ok for me. 
Bash in the shell.

root@pve:/# bash -c "$(wget -qLO - https://raw.githubusercontent.com/oziee/proxmox-ha/main/ha-install.sh)"
[INFO] Using 'sd120gbdiv' for template storage.
[INFO] Using 'nvme0n1' for container storage.
Updating LXC template list...
Creating LXC container...
[INFO] LXC container '126' was successfully created.
Starting LXC container...
Setting up container OS...
Updating container OS...
Installing prerequisites...
Installing Docker...
Configuring Docker...
Configuring NetworkManager...
Creating Home Assistant config...
Downloading Home Assistant Supervisor container...
Installing Home Assistant Supervisor...
Creating service to fix boot time check...
Starting Home Assistant...
Installing the 'ha' cli...
Configuring 'ha' cli prompt...
Cleanup...
[INFO] Successfully created Home Assistant LXC to 126.


Home Assistant is reachable by going to the following URLs.

      http://10.26.9.63:8123
      http://pve.local:8123

**Comment by ghost** [comment]
I am running Prox  7.1-4 and the script still works fine

**Comment by Maltavius** [comment]
Yeah I can run the command as well and get the "Home Assistant is reachable from the following URLs"
But port 8123 isn't actually open.

**Comment by charliesz** [comment]
I'm really new to proxmox and I want now rest homeassistant for the first time. I got the same error using the script as @Hanselljlh in line 131 installing docker.
Could the problem be related to the debian template in proxmox?
Can someone share the used template to validate that? I did not see the script downloading any other template.
Just a few thoughts.

**Comment by marko-p** [comment]
Thanks @oziee, I can confirm your script works with my Proxmox 7.1-4.
One thing I haven't seen mentioned anywhere else is that the directory with ID 'local' has to be set to hold Container templates.

**Comment by Hanselljlh** [comment]
Ok so I don't know why or how but it maybe an issue with hardware. I was on a desktoo with an i9-9700k that it refused to install on. When I got my actual server, R730, the script runs perfectly fine.

**Comment by tteck** [comment]
The HA dev's are making it more difficult to run this type of installation. I'd recommend moving to Home Assistant Container

Scripts For Home Automation
https://tteck.github.io/Proxmox/
